src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/DefaultRazorSemanticTokensInfoService.cs

```diff
@@ -116,8 +114,10 @@ internal class DefaultRazorSemanticTokensInfoService : RazorSemanticTokensInfoSe
             }
 
             var documentVersionStamp = await documentSnapshot.GetTextVersionAsync();
+            var semanticVersion = documentVersionStamp.GetNewerVersion(documentSnapshot.Project.Version);
```

Comment:
Given you do this in two locations I'd just make a local helper method to do this so we're at least guaranteeing the semantic version is calculated the same in both locations. 

Also, even though it is a temporary solution I do have a little bit of concern in comparing project version to document version. If VersionStamp utilizes timestamps under the covers then this is fine; otherwise it's comparing apples and oranges

Reply:
Using a helper now.

Yeah, VersionStamp uses timestamps. One example that got generated was `2021-02-10T19:18:22.8663242Z-10180-0`. BTW, that's what I was talking about yesterday when I thought we might be using VersionStamp wrong, because we already have code where we compare a DocumentVersion and a ProjectVersion or ConfigurationVersion. But once I saw that it's basically just a timestamp I figured that's fine.